libsolidity/AST.cpp

```diff
@@ -337,8 +337,14 @@ void FunctionDefinition::checkTypeRequirements()
 	{
 		if (!var->getType()->canLiveOutsideStorage())
 			BOOST_THROW_EXCEPTION(var->createTypeError("Type is required to live outside storage."));
-		if (!(var->getType()->externalType()) && getVisibility() >= Visibility::Public)
-			BOOST_THROW_EXCEPTION(var->createTypeError("Internal type is not allowed for function with external visibility"));
+		if (getVisibility() >= Visibility::Public && !(var->getType()->externalType()))
+		{
+			// todo delete when will be implemented arrays as parameter type in internal functions
+			if (getVisibility() == Visibility::Public && var->getType()->getCategory() == Type::Category::Array)
+				BOOST_THROW_EXCEPTION(var->createTypeError("Array type is not allowed as parameter for internal functions."));
```

Comment:
Perhaps better: "Arrays only implemented for external functions.", otherwise users might ask why they are not allowed.